FindmallocMC.cmake

```diff
@@ -97,44 +102,47 @@ if(mallocMC_ROOT_DIR)
     # mallocMC variables ########################################################
     #
     set(mallocMC_VERSION "${mallocMC_VERSION_MAJOR}.${mallocMC_VERSION_MINOR}.${mallocMC_VERSION_PATCH}")
-    set(mallocMC_INCLUDE_DIRS ${mallocMC_ROOT_DIR}/include)
+    set(mallocMC_INCLUDE_DIRS ${mallocMC_ROOT_DIR}/include CACHE PATH "mallocMC include pathes")
 
     # check additional components ###############################################
     #
     foreach(COMPONENT ${mallocMC_FIND_COMPONENTS})
-        set(mallocMC_${COMPONENT}_FOUND TRUE)
+        set(mallocMC_${COMPONENT}_FOUND TRUE CACHE BOOL
+            "The mallocMC component ${COMPONENT} exists in the found installation.")
 
         if(${COMPONENT} STREQUAL "halloc")
 
             # halloc linked library #################################################
             #
             list(APPEND mallocMC_REQUIRED_VARS_LIST mallocMC_LIBRARIES)
-            find_library(${COMPONENT}_LIBRARY
+            find_library(mallocMC_${COMPONENT}_LIBRARY
                 NAMES libhalloc.a
                 PATHS "${mallocMC_ROOT_DIR}/../halloc/" ENV HALLOC_ROOT
                 PATH_SUFFIXES "lib" "bin"
-                )
-            if(${COMPONENT}_LIBRARY)
-                list(APPEND mallocMC_LIBRARIES ${${COMPONENT}_LIBRARY})
-            else(${COMPONENT}_LIBRARY)
+                DOC "Libraries for the mallocMC component ${COMPONENT}."
+            )
+            if(mallocMC_${COMPONENT}_LIBRARY)
+                list(APPEND mallocMC_LIBRARIES ${mallocMC_${COMPONENT}_LIBRARY})
+            else(mallocMC_${COMPONENT}_LIBRARY)
                 if(mallocMC_FIND_REQUIRED OR NOT mallocMC_FIND_QUIETLY)
                     message(WARNING "libhalloc.a not found. Ensure it is compiled correctly and set HALLOC_ROOT")
                 endif()
-                set(mallocMC_${COMPONENT}_FOUND FALSE)
-            endif(${COMPONENT}_LIBRARY)
+                unset(mallocMC_${COMPONENT}_FOUND CACHE)
+            endif(mallocMC_${COMPONENT}_LIBRARY)
 
             # halloc headers ########################################################
             #
-            find_path(${COMPONENT}_INCLUDE_DIR
+            find_path(mallocMC_${COMPONENT}_INCLUDE_DIR
                 NAMES halloc.h
                 PATHS "${mallocMC_ROOT_DIR}/../halloc/" ENV HALLOC_ROOT
```

Comment:
I think the real question is "do we want to redistribute *halloc*?".
Currently, the halloc code is not touched / split into policies. So there is no real reason why the whole source code should be shipped with mallocMC directly. Having halloc as an optional dependency would be more suitable. At least that would be the way to handle it if this was about 2 ArchLinux packages ;)